server/plugin.go

```diff
@@ -507,6 +519,16 @@ func (p *Plugin) syncUsers() {
 
 					continue
 				}
+
+				if configuration.AutomaticallyPromoteSyntheticUsers && (mmUser.AuthService != configuration.SyntheticUserAuthService || p.isSyntheticUserAuthDataUpdated) {
```

Comment:
I don't understand the role `isSyntheticUserAuthDataUpdated` plays here. Why isn't the condition that `mmAuthService != configuration.SyntheticUserAuthService` sufficient in this case?

Reply:
The condition `mm.AuthService != configuration.SyntheticUserAuthService` is not sufficient as system admin might change `Auth Data` from, say, `ID` to 'Mail', and we are capturing this change by the variable `isSyntheticUserAuthDataUpdated` as other conditions are not sufficient to capture that change.

Reply:
Ah, that makes sense. Unfortunately, it also means that if this setting is changed while the server is offline (e.g. plugin.json), the job won't be guaranteed to run and update these values.

Can we instead check the `AuthData` for the user here? While we don't get `AuthData` returned from the `GetUsers` call, we do get it from a discrete `GetUser` call and can use that to sidestep `isSyntheticUserAuthDataUpdatedl` altogether.